include/irr/asset/IAssetManager.h

```diff
@@ -31,6 +31,18 @@ namespace asset
     std::function<void(SAssetBundle&)> makeAssetGreetFunc(const IAssetManager* const _mgr);
     std::function<void(SAssetBundle&)> makeAssetDisposeFunc(const IAssetManager* const _mgr);
 
+	//! Class responsible for handling loading of assets from file system or other resources
+	/**
+		It provides the default methods for converting CPU assets to GPU objects, methods 
+		for exporting CPU assets, as well as synthetic asset creators like the IGeometryCreator.
+		There is a loading, writing and creation functionality that is thread-safe.
```

Comment:
Thread safety still has one issue, if you start loading the same asset at the exact same time (they will both check cache and not find anything), you may end up with two copies in the cache.